src/simplewallet/simplewallet.cpp

```diff
@@ -3751,93 +3755,46 @@ bool simple_wallet::init(const boost::program_options::variables_map& vm)
 
     if (m_restoring && m_generate_from_json.empty() && m_generate_from_device.empty())
     {
-      m_wallet->explicit_refresh_from_block_height(!(command_line::is_arg_defaulted(vm, arg_restore_height) ||
-        command_line::is_arg_defaulted(vm, arg_restore_date)));
-      if (command_line::is_arg_defaulted(vm, arg_restore_height) && !command_line::is_arg_defaulted(vm, arg_restore_date))
+      m_wallet->explicit_refresh_from_block_height(!(command_line::is_arg_defaulted(vm, arg_restore_height)));
+      if (!(m_wallet->explicit_refresh_from_block_height()))
       {
         uint16_t year;
         uint8_t month;
         uint8_t day;
-        if (!datestr_to_int(m_restore_date, year, month, day))
-          return false;
-        try
+        
+        std::string electrum_data = std::string(m_electrum_seed.data());
+        std::string seed_date;
+        if (electrum_data.find_first_of("1234567890") != std::string::npos) 
         {
-          m_restore_height = m_wallet->get_blockchain_height_by_date(year, month, day);
```

Comment:
Users must still be able to specify a restore height, if they have an old style seed, they should still be prompted the restore height. Or if they choose to omit their date from their seed, it should ask and store the date they choose into the seed.